crates/pallet-domains/src/lib.rs

```diff
@@ -79,6 +80,7 @@ pub(crate) type NominatorId<T> = <T as frame_system::Config>::AccountId;
 pub trait HoldIdentifier<T: Config> {
     fn staking_staked(operator_id: OperatorId) -> FungibleHoldId<T>;
     fn domain_instantiation_id(domain_id: DomainId) -> FungibleHoldId<T>;
+    fn storage_fund(operator_id: OperatorId) -> FungibleHoldId<T>;
```

Comment:
IIUC, this is not the correct representation. We use this hold when withdrawal is enabled. So the hold reason should have a hint for the user. Something like `storage_fund_withdrawal` would be ideal